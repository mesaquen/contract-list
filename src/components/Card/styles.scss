.card--container {
  margin-top: 0.75rem;
  background-color: var(--white);
  box-shadow: var(--shadow);
  padding: 0.75rem;
}

.card--container-extra-padding {
  padding: 1.5rem;
}

.card--container-round {
  border-radius: var(--roundness);
}
