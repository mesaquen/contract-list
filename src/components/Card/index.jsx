import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './styles.scss'

const Card = ({ children, className, compact, round, ...props }) => {
  const classNames = cx(className, ['card--container'], {
    'card--container-round': round,
    'card--container-extra-padding': !compact,
  })
  return (
    <div {...props} className={classNames}>
      {children}
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  round: PropTypes.bool,
  compact: PropTypes.bool,
}

Card.defaultProps = {
  round: true,
}

export default Card
