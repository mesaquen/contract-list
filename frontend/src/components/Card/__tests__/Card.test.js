import { cleanup, render } from '@testing-library/react'
import Card from '../index'

beforeEach(cleanup)

const renderComponent = (props = {}) => (
  <Card {...props}>
    <span>sample</span>
  </Card>
)

describe('Card', () => {
  it('matches snapshot on compact layout', () => {
    const { asFragment } = render(renderComponent({ compact: true }))
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot on default layout', () => {
    const { asFragment } = render(renderComponent())
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot on square edges', () => {
    const { asFragment } = render(renderComponent({ round: false }))
    expect(asFragment()).toMatchSnapshot()
  })
})
