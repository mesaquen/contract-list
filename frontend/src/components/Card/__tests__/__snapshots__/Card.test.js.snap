// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot on compact layout 1`] = `
<DocumentFragment>
  <div
    class="card--container card--container-round"
  >
    <span>
      sample
    </span>
  </div>
</DocumentFragment>
`;

exports[`Card matches snapshot on default layout 1`] = `
<DocumentFragment>
  <div
    class="card--container card--container-round card--container-extra-padding"
  >
    <span>
      sample
    </span>
  </div>
</DocumentFragment>
`;

exports[`Card matches snapshot on square edges 1`] = `
<DocumentFragment>
  <div
    class="card--container card--container-extra-padding"
  >
    <span>
      sample
    </span>
  </div>
</DocumentFragment>
`;
